packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenReactActivityLifecycleListener.kt

```diff
@@ -0,0 +1,21 @@
+package expo.modules.splashscreen
+
+import android.app.Activity
+import android.content.Context
+import android.os.Bundle
+import android.os.Handler
+import com.facebook.react.ReactRootView
+import expo.modules.splashscreen.singletons.SplashScreen
+import org.unimodules.core.interfaces.ReactActivityLifecycleListener
+
+class SplashScreenReactActivityLifecycleListener(activityContext: Context) : ReactActivityLifecycleListener {
+  override fun onCreate(activity: Activity, savedInstanceState: Bundle?) {
+    // To support backward compatible or SplashScreenImageResizeMode customization
+    // that calling `SplashScreen.show()` in MainActivity,
+    // we postpone the in-module call to the end of main loop.
+    // If MainActivity.onCreate has `SplashScreen.show()`, it will override the call here.
+    Handler(activity.mainLooper).post {
+      SplashScreen.show(activity, SplashScreenImageResizeMode.CONTAIN, ReactRootView::class.java, false)
```

Comment:
As specified earlier, maybe we should read these values from the AndroidManifest.xml or other static resource file in the `android/` folder.

Reply:
yes, plan to have this modularizing expo-splash-screen in another pr.